test/e2e/serial/tests/workload_placement.go

```diff
@@ -802,10 +1008,78 @@ func allocatableResourceType(nrtInfo nrtv1alpha1.NodeResourceTopology, resName c
 	return res.DeepCopy()
 }
 
+// leastAvailableResourceQtyInAllZone allows us to determine the least amount
+// of available resources across all the NRT zones. This is to determine the
+// resources on the zone where they are least present. This can be useful to
+// determine the amount of resources to be provided as a request for a
+// deployment where the number of replicas is the same as the number of
+// NUMA nodes so that all the replicas can successfully obtain resources
+// from all the NUMA nodes.
+func leastAvailableResourceQtyInAllZone(nrtInfo nrtv1alpha1.NodeResourceTopology, baseload baseload.Load, resName corev1.ResourceName) resource.Quantity {
+	var maxResAllocatable resource.Quantity
+
+	// Finding the maximum allocatable resources of a resource type across all zones
+	for _, zone := range nrtInfo.Zones {
+		zoneQty, ok := e2enrt.FindResourceAllocatableByName(zone.Resources, resName.String())
+		if !ok {
+			continue
+		}
+		if zoneQty.Cmp(maxResAllocatable) > 0 {
+			maxResAllocatable = zoneQty
+		}
+	}
+
+	return getLeastAvailableResourceQty(maxResAllocatable, nrtInfo.Zones, resName, baseload)
+}
+
+func getLeastAvailableResourceQty(res resource.Quantity, zones nrtv1alpha1.ZoneList, resName corev1.ResourceName, baseload baseload.Load) resource.Quantity {
+	var zeroVal resource.Quantity
+
+	// We need to take baseload into consideration here. There is no way to
+	// exactly determine how the baseload is distributed across NUMA nodes
+	// so we subtract baseload from both the NUMA nodes to be on the safe side.
+	for _, zone := range zones {
+		zoneQty, ok := e2enrt.FindResourceAvailableByName(zone.Resources, resName.String())
+		if !ok {
+			continue
+		}
+
+		switch resName {
+		case corev1.ResourceCPU:
+			// In case CPU baseload is equal to or greater than the zoneQty, we short circuit to the zero value
+			if zoneQty.Cmp(baseload.CPU()) <= 0 {
+				res = zeroVal
+			}
+			zoneQty.Sub(baseload.CPU())
+
+		case corev1.ResourceMemory:
+			if zoneQty.Cmp(baseload.Memory()) <= 0 {
+				res = zeroVal
+			}
+			zoneQty.Sub(baseload.Memory())
+		}
+		if zoneQty.Cmp(res) < 0 {
+			res = zoneQty
+		}
+	}
+	return res.DeepCopy()
+}
+
 func matchLogLevelToKlog(cnt *corev1.Container, level operatorv1.LogLevel) (bool, bool) {
 	rteFlags := flagcodec.ParseArgvKeyValue(cnt.Args)
 	kLvl := loglevel.ToKlog(level)
 
 	val, found := rteFlags.GetFlag("--v")
 	return found, val.Data == kLvl.String()
 }
+
+func checkReplica(pod corev1.Pod, targetNodeName string, K8sClient *kubernetes.Clientset, podSpec *corev1.PodSpec) {
```

Comment:
`podSpec` is unused (and should not probably be passed by pointer) but we can fix later.